docs/usage/services.rst

```diff
@@ -98,118 +98,118 @@ Complex Operations
-                data["id"] = item_id
-                data = await self.to_model(data, "update")
-                existing_tags = [tag.name for tag in data.tags]
-                tags_to_remove = [tag for tag in data.tags if tag.name not in tags_updated]
-                tags_to_add = [tag for tag in tags_updated if tag not in existing_tags]
+            data_dict = schema_dump(data)
+            tags_updated = data_dict.pop("tags", [])
+
+            # Determine the effective item_id - either from parameter or from the data itself
+            effective_item_id = item_id if item_id is not None else data_dict.get("id")
+
+            # Get existing post to access current tags
+            if effective_item_id is not None:
+                existing_post = await self.get(effective_item_id)
+                existing_tags = [tag.name for tag in existing_post.tags]
+            else:
+                existing_tags = []
+
+            post = await self.to_model(data_dict, "update")
+
+            if tags_updated:
+                tags_updated = [schema_dump(tag) for tag in tags_updated]
+                # Determine tags to remove and add
+                tags_to_remove = [tag for tag in post.tags if tag.name not in [t["name"] for t in tags_updated]]
+                tags_to_add = [tag for tag in tags_updated if tag["name"] not in existing_tags]
+
                 for tag_rm in tags_to_remove:
-                    data.tags.remove(tag_rm)
-                data.tags.extend(
+                    post.tags.remove(tag_rm)
+
+                post.tags.extend(
                     [
-                        await Tag.as_unique_async(self.repository.session, name=tag_text, slug=slugify(tag_text))
-                        for tag_text in tags_to_add
+                        await Tag.as_unique_async(self.repository.session, name=tag["name"], slug=slugify(tag["name"]))
+                        for tag in tags_to_add
                     ],
                 )
-            return await super().update(
-                data=data,
-                item_id=item_id,
-                **kwargs,
-            )
+
+            return await super().update(data=post, item_id=effective_item_id, **kwargs)
 
         # A custom write operation
         async def publish_post(
             self,
             post_id: int,
             publish: bool = True,
         ) -> PostResponse:
-            """Publish or unpublish a post with timestamp."""
-            data = PostUpdate(
-                published=publish,
-                published_at=datetime.datetime.utcnow() if publish else None,
-            )
-            post = await self.repository.update(
+            """Publish or unpublish a post."""
+            post = await self.update(
                 item_id=post_id,
-                data=data,
+                data={"published": publish},
                 auto_commit=True,
             )
             return self.to_schema(post, schema_type=PostResponse)
```

Comment:
What do we want to demonstrate here exactly?  this looks like all the default behavior you can do out of the box.  

We should consider removing it to reduce confusion if there's nothing new.